sdk/eventhub/Azure.Messaging.EventHubs/src/Diagnostics/EventHubsEventSource.cs

```diff
@@ -2450,5 +2680,190 @@ public void UnexpectedException(string errorMessage)
                 WriteEvent(119, identifier ?? string.Empty, eventHubName ?? string.Empty, partitionId ?? string.Empty, operationId ?? string.Empty);
             }
         }
+
+        [NonEvent]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe void WriteEvent<TValue1, TValue2>(int eventId, string arg1, string arg2, TValue1 arg3, TValue2 arg4)
```

Comment:
Due to the nullability checks it is not possible to use this method with TValue being a string.